### PR TITLE
Feature/thruster_offsets

### PR DESCRIPTION
**Pull request by chrstrom**
Adds offsets and arming sequence

### CODE REVIEW

File: firmware/main.c [hunk 1]
```diff
@@ -58,25 +61,38 @@ void I2C_requested()
 	I2C_transmitByte(data);
 }
 
-void arming_check()
+bool arming_check()
 {
-	if (recv_array[0] == ARM_UINT8)
+	bool spin_thrusters = 0;
+	if (recv_array[0] == ARM_UINT8 && recv_array[1] == ARM_UINT8 && recv_array[2] == ARM_UINT8 && recv_array[3] == ARM_UINT8 && recv_array[4] == ARM_UINT8 && recv_array[5] == ARM_UINT8 && recv_array[6] == ARM_UINT8 && recv_array[7] == ARM_UINT8)
```
Comment (oyssolbo) [inline firmware/main.c:67]: Couldn't this be implemented as a for-loop for example?

Comment (oyssolbo) [reply firmware/main.c:67]: Not readable atm (even for embedded)

**Review by oyssolbo — CHANGES_REQUESTED** [review]
Looks good, except one if-sentence

File: firmware/main.c (hunk 1 above)
Comment (chrstrom) [reply firmware/main.c:62]: Nasty hack for pooltest :smile: Everything will be revisited and changed to an FSM-based system after we get this working

File: firmware/main.c [hunk 2]
```diff
@@ -46,12 +46,7 @@ void I2C_received(uint8_t received_data)
 
 	if (num_received_bytes == NUM_THRUSTERS)
 	{
-		if (is_armed)
-		{
-			set_thrusters();
-		}
-		else
-			arming_check();
+		if (arming_check) set_thrusters;
```
Comment (oyssolbo) [inline firmware/main.c:49]: set_thrusters()*

And move to next line

**Review by oyssolbo — CHANGES_REQUESTED** [review]

File: firmware/main.c (hunk 2 above)
Comment (oyssolbo) [reply firmware/main.c:49]: And please move to next line for readability

File: firmware/main.c (hunk 1 above)
Comment (oyssolbo) [reply firmware/main.c:62]: Gotcha

**Review by oyssolbo — APPROVED** [review]

**Review by oyssolbo — APPROVED** [review]

**Pull request merged**